roles/servicetelemetry/templates/manifest_smartgateway_metrics.j2

```diff
@@ -8,6 +8,6 @@ spec:
   amqpDataSource: '{{ agent.name|lower }}'
   debug: false
   serviceType: 'metrics'
-  size: 1
+  size: {% if agent.name == 'collectd' %} {{ smartgateway_deployment_size }} {% else %} 1 {% endif %}
```

Comment:
I believe this is the source of the CI error. It seems to be rendering this template:
```
apiVersion: smartgateway.infra.watch/v2alpha1
kind: SmartGateway
metadata:
  name: 'stf-default-collectd-telemetry'
  namespace: 'service-telemetry'
spec:
  amqpUrl: 'stf-default-interconnect.service-telemetry.svc.cluster.local:5673/collectd/telemetry'
  amqpDataSource: 'collectd'
  debug: false
  serviceType: 'metrics'
  size:  1   prefetch: 15000
  useTimestamp: true
```
This would be because jinja is operating in trim_blocks mode. Looking for the simplest workaround, but might just through a spurious newline in here for now if it works.

Reply:
Try adding a `+` at the end of the `if` statement:

`{% if ... +%}`

This is supposed to disable the `trim_blocks` behaviour at the end of the line. Untested :)

Reply:
Sorry, forgot the ref: https://jinja.palletsprojects.com/en/master/templates/#whitespace-control

Reply:
This LOOKS like the right answer to me, but it isn't working out:

```
 TASK [Generate Smart Gateway manifest and store when override is not provided] ******************************** 
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'template'. Error was a <class 'ansible.errors.AnsibleError'>, original message: t
emplate error while templating string: unexpected 'end of statement block'. String: apiVersion: smartgateway.infra.watch/v2alpha1\nkind: SmartGateway\nmetadata:\n  name: '{{ meta.name }}
-{{ agent.name|lower }}-telemetry'\n  namespace: '{{ meta.namespace }}'\nspec:\n  amqpUrl: '{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local:5673/{{ agent.channel }}'\
n  amqpDataSource: '{{ agent.name|lower }}'\n  debug: false\n  serviceType: 'metrics'\n  size: {% if agent.name == \"collectd\" +%} {{ smartgateway_deployment_size }} {% else %} 1 {% endif %}\n  prefetch: 15000\n  useTimestamp: true\n"}
```

I tried putting it at the end of the `{% endif +%}` but got `template error while templating string: expected token 'end of statement block', got '+'`

The page you link says "Warning: This is the development version. The latest stable version is Version 2.11.x." and if I click through to the 2.11.x it no longer mentions says "Similarly, you can manually disable the trim_blocks behavior by putting a plus sign (+) at the end of a block", instead only mentioning the lstrip suppression by adding a `+` to the beginning. So maybe that feature doesn't exist yet in ansible land. 

Feeling silly, this is so close. Trying `#jinja2:trim_blocks: Flase` on the first line now....

Reply:
What about just building it out on multi line like this instead?

```
{% if ... %}
size: {{ smartgateway_size }}
{% else %}
size: 1
{% endif %}
```

Reply:
That would certainly work and I'll switch to it if the latest thing doesn't. I'm trying to learn something here! ;)

Reply:
I encourage that kind of behaviour, was just giving you the out in case you were done levelling up :)